pkg/clusters/addons/loadimage/addon.go

```diff
@@ -0,0 +1,101 @@
+package loadimage
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"os/exec"
+
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"github.com/kong/kubernetes-testing-framework/pkg/clusters"
+	"github.com/kong/kubernetes-testing-framework/pkg/clusters/types/kind"
+)
+
+// -----------------------------------------------------------------------------
+// CertManager Addon - Builder
+// -----------------------------------------------------------------------------
+
+type Builder struct {
+	image string
+}
+
+func NewBuilder() *Builder {
+	return &Builder{}
+}
+
+func (b *Builder) WithImage(image string) *Builder {
+	b.image = image
+	return b
+}
+
+func (b *Builder) Build() *Addon {
+	return &Addon{
+		image:  b.image,
+		loaded: false,
+	}
+}
+
+// -----------------------------------------------------------------------------
+// CertManager Addon
+// -----------------------------------------------------------------------------
+
+const (
+	// AddonName indicates the unique name of this addon.
+	AddonName clusters.AddonName = "loadimage"
+)
+
+type Addon struct {
+	image  string
+	loaded bool
+}
+
+func New() clusters.Addon {
+	return &Addon{}
+}
+
+// -----------------------------------------------------------------------------
+// CertManager Addon - Addon Implementation
+// -----------------------------------------------------------------------------
+
+func (a *Addon) Name() clusters.AddonName {
+	return AddonName
+}
+
+func (a *Addon) Deploy(ctx context.Context, cluster clusters.Cluster) error {
+	if a.image == "" {
+		a.loaded = true
+		return nil
+	}
+	if cluster.Type() != kind.KindClusterType {
```

Comment:
consider using the Strategy pattern - having several implementations (for KIND, future minikube, etc) implementing a common interface. The concrete implementation to be used could be decided upon by the builder or by `Deploy()` whatever makes more sense.

Reply:
361f92f though with just a switch. I don't see anything obvious we'd add to an interface other than a single "load this" function, whose implementation is opaque to the rest of the addon.

Reply:
If I'm understanding you correctly that you're suggesting we implement the strategy pattern for `Clusters` and to allow some cluster implementations to decide whether they have the capability to use the LoadImage addon, as the inverse to letting the addon make that determination, I do generally like the idea.

However, I'm very hesitant to put the load of doing that on this one addon, especially given that this addon follows an existing precedent, because I think this is something we need to look at holistically and outside of the immediate needs which drove this implementation.

I am OK with us using this simple logic for now, if we create and issue which describes the need @rainest and I have been talking about in zooms recently for clusters to have a system to describe optional cabilities and suggest the strategy pattern there, followed by linking that issue in a comment here and on metallb to point out that there's more TODO here.

Reply:
If I'm understanding you correctly that you're suggesting we implement the strategy pattern for `Clusters` and to allow some cluster implementations to decide whether they have the capability to use the LoadImage addon, as the inverse to letting the addon make that determination, I do generally like the idea.

However, I'm very hesitant to put the load of doing that on this one PR, especially given that this addon follows an existing precedent, because I think this is something we need to look at holistically and outside of the immediate needs which drove this implementation.

I am OK with us using this simple logic for now, if we create and issue which describes the need @rainest and I have been talking about in zooms recently for clusters to have a system to describe optional cabilities and suggest the strategy pattern there, followed by linking that issue in a comment here and on metallb to point out that there's more TODO here.